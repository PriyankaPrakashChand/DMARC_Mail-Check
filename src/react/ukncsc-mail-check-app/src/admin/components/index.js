export { default as AddEntitiesModal } from './AddEntitiesModal';
export { default as DomainAdmin } from './DomainAdmin';
export { default as EntitiesSearch } from './EntitiesSearch';
export { default as EntityAccessGroup } from './EntityAccessGroup';
export { default as EntityDetails } from './EntityDetails';
export { default as EntityDisplay } from './EntityDisplay';
export {
  default as EntitySearchableDropdown,
} from './EntitySearchableDropdown';
export { default as GroupAdmin } from './GroupAdmin';
export { default as UserAdmin } from './UserAdmin';
export { default as withAddRemoveModals } from './withAddRemoveModals';
