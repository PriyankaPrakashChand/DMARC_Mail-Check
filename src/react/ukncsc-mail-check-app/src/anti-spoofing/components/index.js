export { default as DomainsStatusDisplay } from './DomainsStatusDisplay';
export { default as Pagination } from './Pagination';
export { default as RatingDisplay } from './RatingDisplay';
