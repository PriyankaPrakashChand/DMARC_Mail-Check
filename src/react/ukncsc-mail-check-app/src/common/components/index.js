export { default as Breadcrumb } from './Breadcrumb';
export { default as BreadcrumbItem } from './BreadcrumbItem';
export { default as ConfirmationModal } from './ConfirmationModal';
export { default as DetailedTab } from './DetailedTab';
export { default as DetailedTabList } from './DetailedTabList';
export { default as IeFriendlyVictoryChart } from './IeFriendlyVictoryChart';
export { default as MailCheckMessage } from './MailCheckMessage';
export { default as ScrollToTop } from './ScrollToTop';
export { default as ShowMoreDropdown } from './ShowMoreDropdown';
export { default as TermsAndConditions } from './TermsAndConditions';
