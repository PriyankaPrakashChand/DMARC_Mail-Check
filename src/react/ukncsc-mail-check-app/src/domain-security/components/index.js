export { default as AggregateReportChart } from './AggregateReportChart';
export {
  default as AggregateReportChartLegend,
} from './AggregateReportChartLegend';
export {
  default as AggregateReportDateSelector,
} from './AggregateReportDateSelector';
export { default as AggregateReportSummary } from './AggregateReportSummary';
export {
  default as DomainSecurityDetailsMessages,
} from './DomainSecurityDetailsMessages';
export { default as DomainSecurityDetailsMx } from './DomainSecurityDetailsMx';
export {
  default as DomainSecurityDetailsTxt,
} from './DomainSecurityDetailsTxt';
export { default as DomainSecurityPill } from './DomainSecurityPill';
export { default as DomainSecurityRecord } from './DomainSecurityRecord';
export {
  default as DomainSecurityRecordExplanation,
} from './DomainSecurityRecordExplanation';
export { default as DomainSecuritySummary } from './DomainSecuritySummary';
export {
  default as DomainSecuritySummaryMessages,
} from './DomainSecuritySummaryMessages';
export { default as DomainSecuritySummaryMx } from './DomainSecuritySummaryMx';
export {
  default as DomainSecuritySummarySubdomains,
} from './DomainSecuritySummarySubdomains';
export {
  default as DomainSecuritySummaryTxt,
} from './DomainSecuritySummaryTxt';
export { default as DomainSecurityTitle } from './DomainSecurityTitle';
