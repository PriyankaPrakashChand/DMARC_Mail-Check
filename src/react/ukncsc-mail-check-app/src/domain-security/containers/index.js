export {
  default as AggregateReportSummaryContainer,
} from './AggregateReportSummaryContainer';
export {
  default as DomainSecurityAggregateExport,
} from './DomainSecurityAggregateExport';
export {
  default as DomainSecurityDetailsCertificates,
} from './DomainSecurityDetailsCertificates';
export {
  default as DomainSecurityDetailsDkimContainer,
} from './DomainSecurityDetailsDkimContainer';
export {
  default as DomainSecurityDetailsDmarc,
} from './DomainSecurityDetailsDmarc';
export {
  default as DomainSecurityDetailsSpf,
} from './DomainSecurityDetailsSpf';
export {
  default as DomainSecurityDetailsTls,
} from './DomainSecurityDetailsTls';
export {
  default as DomainSecuritySummaryCertificatesContainer,
} from './DomainSecuritySummaryCertificatesContainer';
export {
  default as DomainSecuritySummaryContainer,
} from './DomainSecuritySummaryContainer';
export {
  default as DomainSecuritySummaryDkimContainer,
} from './DomainSecuritySummaryDkimContainer';
export {
  default as DomainSecuritySummaryDmarcContainer,
} from './DomainSecuritySummaryDmarcContainer';
export {
  default as DomainSecuritySummarySpfContainer,
} from './DomainSecuritySummarySpfContainer';
export {
  default as DomainSecuritySummarySubdomainsContainer,
} from './DomainSecuritySummarySubdomainsContainer';
export {
  default as DomainSecuritySummaryTlsContainer,
} from './DomainSecuritySummaryTlsContainer';
export { default as DomainSecurityTlsAdvice } from './DomainSecurityTlsAdvice';
