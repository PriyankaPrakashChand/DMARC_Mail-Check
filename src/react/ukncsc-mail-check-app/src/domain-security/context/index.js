export {
  default as DomainSecurityLocationContext,
} from './DomainSecurityLocationContext';
