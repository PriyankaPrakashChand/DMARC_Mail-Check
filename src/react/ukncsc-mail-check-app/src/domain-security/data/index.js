export * from './domain-security-copy';
export * from './aggregate-report-info-static-data';
export * from './aggregate-report-export-fields';
