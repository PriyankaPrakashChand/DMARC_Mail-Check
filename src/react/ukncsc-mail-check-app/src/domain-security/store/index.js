import { combineReducers } from 'redux';

import aggregateData from './domain-security-aggregate-data';
import certificates from './domain-security-certificates';
import dmarc from './domain-security-dmarc';
import dkim from './domain-security-dkim';
import domain from './domain-security-domain';
import spf from './domain-security-spf';
import subdomains from './domain-security-subdomains';
import tls from './domain-security-tls';

export * from './domain-security-aggregate-data';
export * from './domain-security-certificates';
export * from './domain-security-dkim';
export * from './domain-security-dmarc';
export * from './domain-security-domain';
export * from './domain-security-spf';
export * from './domain-security-subdomains';
export * from './domain-security-tls';

export default combineReducers({
  aggregateData,
  certificates,
  dkim,
  dmarc,
  domain,
  spf,
  subdomains,
  tls,
});
