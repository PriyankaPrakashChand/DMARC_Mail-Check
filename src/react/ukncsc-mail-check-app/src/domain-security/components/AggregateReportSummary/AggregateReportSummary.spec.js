import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';

import { AggregateReportSummary } from 'domain-security/components';

const fetchDomainSecurityAggregateData = jest.fn();
const getDomainSecurityAggregateData = jest.fn();

describe('AggregateReportSummary', () => {
  let container;

  afterEach(() => jest.resetAllMocks());

  describe('when no data returned', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportSummary
          domainId={123}
          fetchDomainSecurityAggregateData={fetchDomainSecurityAggregateData}
          getDomainSecurityAggregateData={getDomainSecurityAggregateData}
        />
      ));
    });

    test('it should show info message', () =>
      expect(
        container.getElementsByClassName('ui info message')[0]
      ).toHaveTextContent(
        'Currently no aggregate report information to show for the selected date range.'
      ));
  });

  describe('when user doesnt have permission', () => {
    beforeEach(() => {
      getDomainSecurityAggregateData.mockImplementation(() => ({
        error: { message: '403 Forbidden' },
      }));

      ({ container } = render(
        <AggregateReportSummary
          domainId={123}
          fetchDomainSecurityAggregateData={fetchDomainSecurityAggregateData}
          getDomainSecurityAggregateData={getDomainSecurityAggregateData}
        />
      ));
    });

    test('it should show info message', () =>
      expect(
        container.getElementsByClassName('ui info message')[0]
      ).toHaveTextContent(
        'You do not have permission to view aggregate reporting for this domain.'
      ));
  });

  describe('when an error occurs', () => {
    beforeEach(() => {
      getDomainSecurityAggregateData.mockImplementation(() => ({
        error: { message: '404 Not found' },
      }));

      ({ container } = render(
        <AggregateReportSummary
          domainId={123}
          fetchDomainSecurityAggregateData={fetchDomainSecurityAggregateData}
          getDomainSecurityAggregateData={getDomainSecurityAggregateData}
        />
      ));
    });

    test('it should show error message', () =>
      expect(
        container.getElementsByClassName('ui error message')[0]
      ).toHaveTextContent(
        'There was a problem retrieving aggregate report information for this domain: '
      ));
  });

  describe('when successfully retrieves data', () => {
    beforeEach(() => {
      getDomainSecurityAggregateData.mockImplementation(() => ({
        data: {
          results: {
            '2017-12-10T00:00:00': {
              fullyTrusted: 4,
              partiallyTrusted: 6,
              untrusted: 0,
              quarantined: 0,
              rejected: 0,
            },
            '2017-12-11T00:00:00': {
              fullyTrusted: 5,
              partiallyTrusted: 8,
              untrusted: 0,
              quarantined: 0,
              rejected: 0,
            },
            '2017-12-12T00:00:00': {
              fullyTrusted: 6,
              partiallyTrusted: 7,
              untrusted: 0,
              quarantined: 0,
              rejected: 0,
            },
          },
          totalEmail: 178,
        },
      }));

      ({ container } = render(
        <AggregateReportSummary
          domainId={123}
          fetchDomainSecurityAggregateData={fetchDomainSecurityAggregateData}
          getDomainSecurityAggregateData={getDomainSecurityAggregateData}
        />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
