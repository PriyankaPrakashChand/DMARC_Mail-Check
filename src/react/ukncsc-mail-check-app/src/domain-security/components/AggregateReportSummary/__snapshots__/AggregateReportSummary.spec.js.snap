// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregateReportSummary when successfully retrieves data it should match the snapshot 1`] = `
<div>
  <form
    class="ui form"
  >
    <div
      style="padding-bottom: 20px;"
      class="ui stackable grid"
    >
      <div
        class="row"
      >
        <div
          class="nine wide column"
        >
          <div
            class="field AggregateReportSummary--date-range"
          >
            <div
              role="listbox"
              aria-expanded="false"
              class="ui dropdown"
              tabindex="0"
            >
              <div
                class="text"
                role="alert"
                aria-live="polite"
              />
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              >
                <div
                  style="pointer-events: all;"
                  role="option"
                  aria-checked="false"
                  aria-selected="true"
                  class="selected item"
                >
                  <span
                    class="text"
                  >
                    Last 7 days
                  </span>
                </div>
                <div
                  style="pointer-events: all;"
                  role="option"
                  aria-checked="false"
                  aria-selected="false"
                  class="item"
                >
                  <span
                    class="text"
                  >
                    Last 14 days
                  </span>
                </div>
                <div
                  style="pointer-events: all;"
                  role="option"
                  aria-checked="false"
                  aria-selected="false"
                  class="item"
                >
                  <span
                    class="text"
                  >
                    Last 30 days
                  </span>
                </div>
                <div
                  style="pointer-events: all;"
                  role="option"
                  aria-checked="false"
                  aria-selected="false"
                  class="item"
                >
                  <span
                    class="text"
                  >
                    Last 3 months
                  </span>
                </div>
                <div
                  style="pointer-events: all;"
                  role="option"
                  aria-checked="false"
                  aria-selected="false"
                  class="item"
                >
                  <span
                    class="text"
                  >
                    Last 6 months
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="three wide column"
        >
          <div
            class="field"
          >
            <div
              class="ui checked toggle checkbox"
            >
              <input
                class="hidden"
                readonly=""
                tabindex="0"
                type="checkbox"
                value=""
                checked=""
              />
              <label>
                Include subdomains
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <div>
    <div
      class="ui small statistic"
    >
      <div
        style="font-weight: 400;"
        class="label"
      >
        Total emails:
      </div>
      <div
        style="font-weight: 500;"
        class="value"
      >
        178.0
      </div>
    </div>
  </div>
  <div
    class="ui stackable grid"
  >
    <div
      class="row"
    >
      <div
        style="padding: 0px;"
        class="ten wide column"
      >
        <div
          style="position: relative; height: 0px; width: 100%; padding: 0px 0px 57.14285714285714% 0px;"
        >
          <div
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; pointer-events: none;"
            class="VictoryContainer"
          >
            <svg
              width="700"
              height="400"
              role="img"
              aria-labelledby="victory-container-1-title victory-container-1-desc"
              viewBox="0 0 700 400"
              style="pointer-events: all; width: 100%; height: 100%;"
            >
              <g
                role="presentation"
              >
                <line
                  x1="50"
                  x2="50"
                  y1="50"
                  y2="350"
                  style="stroke: #252525; stroke-width: 0; fill: transparent; stroke-linecap: round; stroke-linejoin: round;"
                  role="presentation"
                  shape-rendering="auto"
                  vector-effect="non-scaling-stroke"
                />
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="650"
                    y1="303.84615384615387"
                    y2="303.84615384615387"
                    style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="50"
                    x2="49"
                    y1="303.84615384615387"
                    y2="303.84615384615387"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="39"
                    dx="0"
                    y="303.84615384615387"
                    dy="4.26"
                  >
                    <tspan
                      x="39"
                      dx="0"
                      text-anchor="end"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      2.0
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="650"
                    y1="257.6923076923077"
                    y2="257.6923076923077"
                    style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="50"
                    x2="49"
                    y1="257.6923076923077"
                    y2="257.6923076923077"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="39"
                    dx="0"
                    y="257.6923076923077"
                    dy="4.26"
                  >
                    <tspan
                      x="39"
                      dx="0"
                      text-anchor="end"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      4.0
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="650"
                    y1="211.53846153846152"
                    y2="211.53846153846152"
                    style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="50"
                    x2="49"
                    y1="211.53846153846152"
                    y2="211.53846153846152"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="39"
                    dx="0"
                    y="211.53846153846152"
                    dy="4.26"
                  >
                    <tspan
                      x="39"
                      dx="0"
                      text-anchor="end"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      6.0
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="650"
                    y1="165.3846153846154"
                    y2="165.3846153846154"
                    style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="50"
                    x2="49"
                    y1="165.3846153846154"
                    y2="165.3846153846154"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="39"
                    dx="0"
                    y="165.3846153846154"
                    dy="4.26"
                  >
                    <tspan
                      x="39"
                      dx="0"
                      text-anchor="end"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      8.0
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="650"
                    y1="119.23076923076923"
                    y2="119.23076923076923"
                    style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="50"
                    x2="49"
                    y1="119.23076923076923"
                    y2="119.23076923076923"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="39"
                    dx="0"
                    y="119.23076923076923"
                    dy="4.26"
                  >
                    <tspan
                      x="39"
                      dx="0"
                      text-anchor="end"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      10.0
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="650"
                    y1="73.07692307692304"
                    y2="73.07692307692304"
                    style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="50"
                    x2="49"
                    y1="73.07692307692304"
                    y2="73.07692307692304"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="39"
                    dx="0"
                    y="73.07692307692304"
                    dy="4.26"
                  >
                    <tspan
                      x="39"
                      dx="0"
                      text-anchor="end"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      12.0
                    </tspan>
                  </text>
                </g>
              </g>
              <g
                role="presentation"
              >
                <line
                  x1="50"
                  x2="650"
                  y1="350"
                  y2="350"
                  style="stroke: #c1c2c4; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                  role="presentation"
                  shape-rendering="auto"
                  vector-effect="non-scaling-stroke"
                />
                <g
                  role="presentation"
                >
                  <line
                    x1="50"
                    x2="50"
                    y1="350"
                    y2="50"
                    style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="50"
                    x2="50"
                    y1="350"
                    y2="351"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="50"
                    dx="0"
                    y="361"
                    dy="10.26"
                  >
                    <tspan
                      x="50"
                      dx="0"
                      text-anchor="middle"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      10th Dec 17
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    x1="350"
                    x2="350"
                    y1="350"
                    y2="50"
                    style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="350"
                    x2="350"
                    y1="350"
                    y2="351"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="350"
                    dx="0"
                    y="361"
                    dy="10.26"
                  >
                    <tspan
                      x="350"
                      dx="0"
                      text-anchor="middle"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      11th Dec 17
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    x1="650"
                    x2="650"
                    y1="350"
                    y2="50"
                    style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <line
                    x1="650"
                    x2="650"
                    y1="350"
                    y2="351"
                    style="stroke: transparent; fill: transparent; size: 1px;"
                    role="presentation"
                    shape-rendering="auto"
                    vector-effect="non-scaling-stroke"
                  />
                  <text
                    x="650"
                    dx="0"
                    y="361"
                    dy="10.26"
                  >
                    <tspan
                      x="650"
                      dx="0"
                      text-anchor="middle"
                      style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                    >
                      12th Dec 17
                    </tspan>
                  </text>
                </g>
              </g>
              <g>
                <g
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                >
                  <defs>
                    <clippath
                      id="victory-clip-2"
                    >
                      <rect
                        x="50"
                        y="50"
                        width="600"
                        height="300"
                        role="presentation"
                        shape-rendering="auto"
                        vector-effect="non-scaling-stroke"
                      />
                    </clippath>
                  </defs>
                  <g
                    clip-path="url(#victory-clip-2)"
                  >
                    <g>
                      <path
                        d="M50,350L350,350L650,350L650,350L350,350L50,350Z"
                        style="fill: #f0a1f1; stroke: none; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                      <path
                        d="M50,350L350,350L650,350"
                        style="fill: none; stroke: #3a0936; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                    </g>
                  </g>
                </g>
                <g
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                >
                  <defs>
                    <clippath
                      id="victory-clip-3"
                    >
                      <rect
                        x="50"
                        y="50"
                        width="600"
                        height="300"
                        role="presentation"
                        shape-rendering="auto"
                        vector-effect="non-scaling-stroke"
                      />
                    </clippath>
                  </defs>
                  <g
                    clip-path="url(#victory-clip-3)"
                  >
                    <g>
                      <path
                        d="M50,350L350,350L650,350L650,350L350,350L50,350Z"
                        style="fill: #f5c05e; stroke: none; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                      <path
                        d="M50,350L350,350L650,350"
                        style="fill: none; stroke: #221803; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                    </g>
                  </g>
                </g>
                <g
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                >
                  <defs>
                    <clippath
                      id="victory-clip-4"
                    >
                      <rect
                        x="50"
                        y="50"
                        width="600"
                        height="300"
                        role="presentation"
                        shape-rendering="auto"
                        vector-effect="non-scaling-stroke"
                      />
                    </clippath>
                  </defs>
                  <g
                    clip-path="url(#victory-clip-4)"
                  >
                    <g>
                      <path
                        d="M50,350L350,350L650,350L650,350L350,350L50,350Z"
                        style="fill: #9fd0fb; stroke: none; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                      <path
                        d="M50,350L350,350L650,350"
                        style="fill: none; stroke: #071f3d; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                    </g>
                  </g>
                </g>
                <g
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                >
                  <defs>
                    <clippath
                      id="victory-clip-5"
                    >
                      <rect
                        x="50"
                        y="50"
                        width="600"
                        height="300"
                        role="presentation"
                        shape-rendering="auto"
                        vector-effect="non-scaling-stroke"
                      />
                    </clippath>
                  </defs>
                  <g
                    clip-path="url(#victory-clip-5)"
                  >
                    <g>
                      <path
                        d="M50,211.53846153846152L350,165.3846153846154L650,188.46153846153848L650,350L350,350L50,350Z"
                        style="fill: #98f0e9; stroke: none; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                      <path
                        d="M50,211.53846153846152L350,165.3846153846154L650,188.46153846153848"
                        style="fill: none; stroke: #0b2321; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                    </g>
                  </g>
                </g>
                <g
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
                >
                  <defs>
                    <clippath
                      id="victory-clip-6"
                    >
                      <rect
                        x="50"
                        y="50"
                        width="600"
                        height="300"
                        role="presentation"
                        shape-rendering="auto"
                        vector-effect="non-scaling-stroke"
                      />
                    </clippath>
                  </defs>
                  <g
                    clip-path="url(#victory-clip-6)"
                  >
                    <g>
                      <path
                        d="M50,119.23076923076923L350,50L650,50L650,188.46153846153848L350,165.3846153846154L50,211.53846153846152Z"
                        style="fill: #c6cd80; stroke: none; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                      <path
                        d="M50,119.23076923076923L350,50L650,50"
                        style="fill: none; stroke: #1a1e04; stroke-width: 2;"
                        role="presentation"
                        shape-rendering="auto"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </svg>
            <div
              style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
            >
              <svg
                width="700"
                height="400"
                viewBox="0 0 700 400"
                style="overflow: visible; width: 100%; height: 100%;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        style="padding: 0px;"
        class="bottom aligned two wide column"
      >
        <ul
          class="ChartLegend"
        >
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(198, 205, 128); stroke: #1a1e04;"
            />
            Fully trusted
          </li>
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(152, 240, 233); stroke: #0b2321;"
            />
            Partially trusted
          </li>
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(159, 208, 251); stroke: #071f3d;"
            />
            Untrusted
          </li>
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(245, 192, 94); stroke: #221803;"
            />
            Quarantined
          </li>
          <li>
            <h2
              class="ui header ChartLegendHeader"
            >
              -
            </h2>
            <div
              class="ChartLegendMarker"
              style="background: rgb(240, 161, 241); stroke: #3a0936;"
            />
            Rejected
          </li>
        </ul>
      </div>
    </div>
    <div
      class="five column row"
    >
      <div
        class="column"
      >
        <h5
          class="ui header"
        >
          Fully trusted
        </h5>
        <p>
          Emails delivered with both successful SPF and DKIM.
        </p>
      </div>
      <div
        class="column"
      >
        <h5
          class="ui header"
        >
          Partially trusted
        </h5>
        <p>
          Emails delivered with either successful SPF or DKIM.
        </p>
      </div>
      <div
        class="column"
      >
        <h5
          class="ui header"
        >
          Untrusted
        </h5>
        <p>
          Emails delivered where SPF and DKIM both failed.
        </p>
      </div>
      <div
        class="column"
      >
        <h5
          class="ui header"
        >
          Quarantined
        </h5>
        <p>
          Emails sent to spam folder because both SPF and DKIM failed.
        </p>
      </div>
      <div
        class="column"
      >
        <h5
          class="ui header"
        >
          Rejected
        </h5>
        <p>
          Emails not delivered to the user because both SPF and DKIM failed.
        </p>
      </div>
    </div>
  </div>
  <div
    class="ui hidden divider"
  />
</div>
`;
